packages/bun-usockets/src/crypto/openssl.c

```diff
@@ -363,8 +369,9 @@ struct us_internal_ssl_socket_t *ssl_on_data(struct us_internal_ssl_socket_t *s,
   loop_ssl_data->ssl_socket = &s->s;
   loop_ssl_data->msg_more = 0;
 
-  if (us_socket_is_closed(0, &s->s)) {
-    return s;
+  if (!s || !s->ssl || us_socket_is_closed(0, &s->s) ||
```

Comment:
How is this not a use after free? If !s ever happens this would crash on line 369.